pkg/download/classic/download.go

```diff
@@ -56,92 +51,79 @@ func Download(client dtclient.Client, projectName string, apisToDownload api.API
-			for _, c := range downloadedConfigs {
-				content, err := c.Template.Content()
-				if err != nil {
-					return
-				}
-				if shouldPersist(currentApi, content, filters) {
-					configsToPersist = append(configsToPersist, c)
-				} else {
-					lg.Debug("\tSkipping persisting config %v (%v) in API %v", c.value.Id, c.value.Name, currentApi.ID)
-				}
+
+			foundValues, err := findConfigsToDownload(client, currentApi, filters)
+			if err != nil {
+				log.WithFields(field.Error(err), field.Type(currentApi.ID)).Error("Failed to fetch configs of type '%s', skipping download of this type. Reason: %v", currentApi.ID, err)
+				return
 			}
-			if len(configsToPersist) > 0 {
+
+			foundValues = filterConfigsToSkip(currentApi, foundValues, filters)
+			if len(foundValues) == 0 {
+				log.WithFields(field.Type(currentApi.ID)).Debug("No configs of type '%s' to download", currentApi.ID)
+				return
+			}
+
+			log.WithFields(field.Type(currentApi.ID)).Debug("Found %d configs of type '%s' to download", len(foundValues), currentApi.ID)
+			if configs := downloadConfigs(client, currentApi, foundValues, projectName, filters); len(configs) > 0 {
 				mutex.Lock()
-				results[currentApi.ID] = getConfigsFromCustomConfigs(configsToPersist)
+				results[currentApi.ID] = configs
 				mutex.Unlock()
 			}
 		}()
 	}
-	log.Debug("Started all downloads")
 	wg.Wait()
-
 	duration := time.Since(startTime).Truncate(1 * time.Second)
 	log.Debug("Finished fetching all configs in %v", duration)
 
 	return results, nil
 }
 
-func getConfigsFromCustomConfigs(customConfigs []downloadedConfig) []config.Config {
-	var finalConfigs []config.Config
-	for _, c := range customConfigs {
-		finalConfigs = append(finalConfigs, c.Config)
-	}
-	return finalConfigs
-}
-
-func downloadConfigs(client dtclient.Client, api api.API, projectName string, filters ContentFilters) []downloadedConfig {
-	var results []downloadedConfig
-	logger := log.WithFields(field.Type(api.ID))
-	foundValues, err := findConfigsToDownload(client, api, filters)
-	if err != nil {
-		logger.WithFields(field.Error(err)).Error("Failed to fetch configs of type '%v', skipping download of this type. Reason: %v", api.ID, err)
-		return results
-	}
-
-	foundValues = filterConfigsToSkip(api, foundValues, filters)
-	if len(foundValues) == 0 {
-		logger.Debug("No configs of type '%v' to download", api.ID)
-		return results
-	}
-
-	logger.Debug("Found %d configs of type %q to download", len(foundValues), api.ID)
+func downloadConfigs(client dtclient.Client, api api.API, configsToDownload values, projectName string, filters ContentFilters) []config.Config {
+	var results []config.Config
 
 	mutex := sync.Mutex{}
 	wg := sync.WaitGroup{}
-	wg.Add(len(foundValues))
-	for _, v := range foundValues {
+	wg.Add(len(configsToDownload))
+	for _, v := range configsToDownload {
 		go func() {
 			defer wg.Done()
 
-			downloadedJsons, err := downloadAndUnmarshalConfig(client, api, v)
+			dlData, err := download(client, api, v)
```

Comment:
You didn't introduce this @warber, but: we capture the config to download `v` in the loop and go routine, rather than passing it as a variable to the go func. This seems not to have caused issues so far - but couldn't it?

Reply:
Isn't that only a problem for go <1.22 ?